include/zephyr/net/net_mgmt.h

```diff
@@ -36,38 +35,28 @@ struct net_if;
 /**
  * @brief NET MGMT event mask basics, normalizing parts of bit fields
  */
-#define NET_MGMT_EVENT_MASK		0x80000000
-#define NET_MGMT_ON_IFACE_MASK		0x40000000
-#define NET_MGMT_LAYER_MASK		0x30000000
-#define NET_MGMT_SYNC_EVENT_MASK	0x08000000
-#define NET_MGMT_LAYER_CODE_MASK	0x07FF0000
-#define NET_MGMT_COMMAND_MASK		0x0000FFFF
-
-#define NET_MGMT_EVENT_BIT		BIT(31)
-#define NET_MGMT_IFACE_BIT		BIT(30)
-#define NET_MGMT_SYNC_EVENT_BIT		BIT(27)
-
-#define NET_MGMT_LAYER(_layer)		(_layer << 28)
-#define NET_MGMT_LAYER_CODE(_code)	(_code << 16)
-
-#define NET_MGMT_EVENT(mgmt_request)		\
-	(mgmt_request & NET_MGMT_EVENT_MASK)
+#define NET_MGMT_EVENT_MASK        GENMASK64(63, 63) /* 0x8000000000000000 */
+#define NET_MGMT_ON_IFACE_MASK     GENMASK64(62, 62) /* 0x4000000000000000 */
+#define NET_MGMT_LAYER_MASK        GENMASK64(61, 60) /* 0x3000000000000000 */
+#define NET_MGMT_SYNC_EVENT_MASK   GENMASK64(59, 59) /* 0x0800000000000000 */
+#define NET_MGMT_LAYER_CODE_MASK   GENMASK64(58, 52) /* 0x07F0000000000000 */
+#define NET_MGMT_COMMAND_MASK      GENMASK64(51, 0)  /* 0x000FFFFFFFFFFFFF */
 
-#define NET_MGMT_ON_IFACE(mgmt_request)		\
-	(mgmt_request & NET_MGMT_ON_IFACE_MASK)
+#define NET_MGMT_MAX_COMMANDS 52 /* TODO: figure out the value from mask */
 
-#define NET_MGMT_EVENT_SYNCHRONOUS(mgmt_request)	\
-	(mgmt_request & NET_MGMT_SYNC_EVENT_MASK)
+#define NET_MGMT_EVENT_BIT         BIT64(63)
+#define NET_MGMT_IFACE_BIT         BIT64(62)
+#define NET_MGMT_SYNC_EVENT_BIT    BIT64(59)
 
-#define NET_MGMT_GET_LAYER(mgmt_request)	\
-	((mgmt_request & NET_MGMT_LAYER_MASK) >> 28)
-
-#define NET_MGMT_GET_LAYER_CODE(mgmt_request)	\
-	((mgmt_request & NET_MGMT_LAYER_CODE_MASK) >> 16)
-
-#define NET_MGMT_GET_COMMAND(mgmt_request)	\
-	(mgmt_request & NET_MGMT_COMMAND_MASK)
+#define NET_MGMT_LAYER(_layer)     FIELD_PREP(NET_MGMT_LAYER_MASK, (_layer))
+#define NET_MGMT_LAYER_CODE(_code) FIELD_PREP(NET_MGMT_LAYER_CODE_MASK, (_code))
 
+#define NET_MGMT_EVENT(mgmt_request)             FIELD_GET(NET_MGMT_EVENT_MASK, mgmt_request)
+#define NET_MGMT_ON_IFACE(mgmt_request)          FIELD_GET(NET_MGMT_ON_IFACE_MASK, mgmt_request)
+#define NET_MGMT_EVENT_SYNCHRONOUS(mgmt_request) FIELD_GET(NET_MGMT_SYNC_EVENT_MASK, mgmt_request)
+#define NET_MGMT_GET_LAYER(mgmt_request)         FIELD_GET(NET_MGMT_LAYER_MASK, mgmt_request)
+#define NET_MGMT_GET_LAYER_CODE(mgmt_request)    FIELD_GET(NET_MGMT_LAYER_CODE_MASK, mgmt_request)
+#define NET_MGMT_GET_COMMAND(mgmt_request)       FIELD_GET(NET_MGMT_COMMAND_MASK, mgmt_request)
```

Comment:
Will this return a bitmask? I'm concerned there would be some confusion about this, event codes are still expressed by integers (`NET_EVENT_IF_CMD_DOWN` etc), but inside the event itself (`NET_EVENT_IF_DOWN`) it's converted to bitmask. I saw that for example `conn_mgr` uses `NET_MGMT_GET_COMMAND()` and compares it directly with event code (https://github.com/zephyrproject-rtos/zephyr/blob/c9f149cf6f445d7e39681ed0509e7acab5dee25b/subsys/net/conn_mgr/events_handler.c#L40-L41)

Reply:
Note that I only changed `tests/net/mgmt` tests to pass (at least the default test), so there are probably lot of things to change in the code still if we want to go with these changes.

Reply:
The reason why that conn_mgr example would not work is that `NET_MGMT_GET_COMMAND()` would return the bits only (not the layer), but the actual event `NET_EVENT_IF_CMD_DOWN` will also contain the layer value.

Reply:
Fixed this issue in the latest version, for example the `NET_EVENT_IF_CMD_DOWN` is now a bit field so that the macro `NET_MGMT_GET_COMMAND()` can return it correctly.